www/js/callbacks.js

```diff
@@ -1019,6 +1019,54 @@ Callbacks = {
         CSEMOTELIST.handleChange();
     },
 
+    renameEmote: function (data) {
+        var badBefore = /\s/g.test(data.old);
+        var badAfter = /\s/g.test(data.name);
+        var oldName = data.old;
+        delete data.old;
+
+        data.regex = new RegExp(data.source, "gi");
+
+        for (var i = 0; i < CHANNEL.emotes.length; i++) {
+            if (CHANNEL.emotes[i].name === oldName) {
+                CHANNEL.emotes[i] = data;
+                break;
+            }
+        }
+
+        // Now bad
+        if(badAfter){
+            // But wasn't bad before: Add it to bad list
```

Comment:
I've been meaning to separate out all the emote logic into a shared interface that can be implemented by the old method, and the new method, and then phase the old one out.  That would simplify this a lot.